pulsar/consumer_partition.go

```diff
@@ -220,10 +230,19 @@ func (pc *partitionConsumer) MessageReceived(response *pb.CommandMessage, header
 	reader := internal.NewMessageReader(headersAndPayload)
 	msgMeta, err := reader.ReadMessageMetadata()
 	if err != nil {
-		// TODO send discardCorruptedMessage
+		pc.discardCorruptedMessage(pbMsgID, pb.CommandAck_ChecksumMismatch)
 		return err
 	}
 
+	uncompressedHeadersAndPayload, err := pc.Decompress(msgMeta, headersAndPayload)
```

Comment:
is there a way to implement this logic in the message reader?

Reply:
Main reason to have this outside is to handle the proper "decompress error" without the need to check the error types.